### ISSUE TITLE
Messages not settled in ModeFirst

### ISSUE THREAD

**Issue by eljoth** [Azure/go-amqp#35]
Hello,

I am connecting in modefirst with a linkcredit of 100 as follows:

```go
err := q.receiver.HandleMessage(ctx, q.onPublish)
if err != nil {
q.log.Error(err, "Error with queue")
}
```

```go
func (q *Queue) onPublish(msg *amqp.Message) error {
	message := new(core.Message)
	if err := json.Unmarshal(msg.GetData(), message); err != nil {
		q.log.Error(err, "could not get data")
		msg.Reject(context.Background(), &amqp.Error{
			Condition:   "",
			Description: err.Error(),
			Info:        nil,
		})
		return err
	}
	q.messages <- *message
	err := msg.Accept(context.Background())
	if err != nil {
		return err
	}
	return nil
}
```

After hitting the credit of 100 messages, my service does not receive messages anymore although the broker says, that 100 messages were acknowledget by the subscriber.

I assume that go-amqp with the new `HandleMessage` doesn't differ that much from the introduction example, but what am I doing wrong ?

Best regards

**Comment by philschleier** [comment]
Hi @eljoth , this looks like #33 to me? Messages get added to the list of unsettled messages, but because you're in ModeFirst they are never removed.

**Comment by haraldkuegel** [comment]
Hello,

i am facing the same problem. I tried various settings of LinkOptions, but none worked for me (maybe except for setting SessionIncomingWindow to maxInt). It would be great to have little guide how to set the LinkOptions, especially if the default settings don't work right away.

Thanks!

**Comment by jhendrixMSFT** [comment]
Fixed in v0.16.0
